api/app/src/test/kotlin/packit/integration/controllers/LoginControllerTest.kt

```diff
@@ -61,12 +61,91 @@ class LoginControllerTestGithub : IntegrationTest()
     }
 
     @Test
-    fun `returns forbidden when basic login is disabled`()
+    fun `basic login returns forbidden when basic login is disabled`()
     {
         val result =
             LoginTestHelper.getBasicLoginResponse(LoginWithPassword("email@email.com", "password"), restTemplate)
         assertForbidden(result)
     }
+
+    @Test
+    fun `preauth login returns forbidden when basic login is disabled`()
+    {
+        val result =
+            LoginTestHelper.getPreauthLoginResponse(
+                "preauth.user", "Preauth User", "preauth.user@example.com",
+                restTemplate
+            )
+        assertForbidden(result)
+    }
+}
+
+@TestPropertySource(properties = ["auth.method=preauth"])
+class LoginControllerTestPreAuth : IntegrationTest()
+{
+    private val userEmail = "test.user@example.com"
+    private val userName = "test.user"
+    private val userDisplayName = "Test User"
+
+    @Autowired
+    lateinit var userRepository: UserRepository
+
+    @AfterEach
+    fun cleanupData()
+    {
+        if (userRepository.existsByEmail(userEmail)) {
+            userRepository.deleteByEmail(userEmail)
+        }
+    }
+
+    @Test
+    fun `can login without existing user`()
+    {
+        val result =
+            packit.integration.controllers.LoginTestHelper.getPreauthLoginResponse(
+                userName, userDisplayName, userEmail, restTemplate
+            )
+
+        assertSuccess(result)
+        val packitToken = jacksonObjectMapper().readTree(result.body).get("token").asText()
+        assertThat(packitToken.count()).isGreaterThan(0)
+        assertThat(userRepository.existsByUsername(userName)).isTrue()
+    }
+
+    @Test
+    fun `can login with existing user`()
+    {
+        val testUser = User(
+            username = userName,
+            displayName = userDisplayName,
+            disabled = false,
+            email = userEmail,
+            userSource = "preauth",
+            roles = mutableListOf(),
+            password = null,
+            lastLoggedIn = null
+        )
+        userRepository.save(testUser)
+
+        val result =
+            packit.integration.controllers.LoginTestHelper.getPreauthLoginResponse(
+                userName, userDisplayName, userEmail, restTemplate
+            )
+
+        assertSuccess(result)
+        val packitToken = jacksonObjectMapper().readTree(result.body).get("token").asText()
+        assertThat(packitToken.count()).isGreaterThan(0)
+    }
+
+    @Test
+    fun `returns 400 when username header not provided`()
+    {
+        val result =
+            packit.integration.controllers.LoginTestHelper.getPreauthLoginResponse(
+                null, userDisplayName, userEmail, restTemplate
+            )
+        assertEquals(result.statusCode, HttpStatus.BAD_REQUEST)
```

Comment:
```suggestion
        assertEquals(result.statusCode, HttpStatus.BAD_REQUEST)
        assertThat(packitToken.count()).isEqualTo(0)
```

Reply:
It won't be an empty list, but I'll test that a token node isn't in the response body when there's an error.